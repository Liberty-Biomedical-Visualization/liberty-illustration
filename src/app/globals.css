@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --accent-50: 224 247 252;
  --accent-100: 178 224 240;
  --accent-200: 128 200 232;
  --accent-300: 77 176 215;
  --accent-400: 26 153 206;
  --accent-500: 3 152 206;
  --accent-600: 2 128 165;
  --accent-700: 2 109 136;
  --accent-800: 2 90 107;
  --accent-900: 2 61 77;
  --accent-950: 1 44 58;
  --background: 255 255 255;
  --foreground: 23 23 23;
}

body {
  color: var(--foreground);
  background: var(--background);
  font-family: Arial, Helvetica, sans-serif;
}

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}
