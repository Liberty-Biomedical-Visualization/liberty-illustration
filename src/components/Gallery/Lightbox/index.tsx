import { faXmark } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import FocusTrap from "focus-trap-react";
import { MutableRefObject, useEffect, useRef, useState } from "react";

import AccessiblyHiddenText from "@/components/AccessiblyHiddenText";
import Carousel, { type CarouselProps } from "@/components/Carousel";
import type { ImageGallery } from "@/lib/content";
import resolveClassNames from "@/lib/resolveClassNames";
import {
  type TransitionDuration,
  transitionDurationClassNames,
} from "@/lib/TransitionDuration";

export default function Lightbox(props: Readonly<LightboxProps>) {
  const {
    disableTabbableDisplayCheck,
    imageIndex,
    images,
    imageTransitionDuration,
    onClose,
    visibilityTransitionDuration,
  } = props;

  const timeoutId = useRef<number | null>(null);
  useEffect(() => cleanUpPendingTimeout(timeoutId), []);

  useEffect(() => disableScrolling(), []);

  const isInitiallyVisible = visibilityTransitionDuration === null;
  const [isVisible, setIsVisible] = useState(isInitiallyVisible);
  useEffect(
    () => makeVisible(isInitiallyVisible, setIsVisible),
    [isInitiallyVisible],
  );

  const opacity = isVisible ? "opacity-100" : "opacity-0";

  const transition =
    visibilityTransitionDuration === null
      ? ""
      : `${transitionDurationClassNames[visibilityTransitionDuration]} transition`;

  const className = resolveClassNames(
    "bg-foreground/95 fixed flex flex-col inset-0 items-center justify-center overscroll-contain px-6 py-10 sm:px-14 z-30",
    opacity,
    transition,
    props.className,
  );

  const orderedImages = [
    ...images.slice(imageIndex),
    ...images.slice(0, imageIndex),
  ];

  const sizedImages = orderedImages.map(transformToSizedImage);

  const handleClose = () => {
    if (isInitiallyVisible) {
      onClose();
    } else {
      setIsVisible(false);
      timeoutId.current = window.setTimeout(
        onClose,
        visibilityTransitionDuration,
      );
    }
  };

  const handleEscape = () => {
    handleClose();
    // Returning false prevents the FocusTrap from being deactivated immediately
    // when escape is pressed. This gives control over the FocusTrap to
    // Lightboxâ€™s parent by dismounting the component in onClose.
    return false;
  };

  return (
    <FocusTrap
      focusTrapOptions={{
        escapeDeactivates: handleEscape,
        tabbableOptions: {
          // Setting displayCheck to "none" allows FocusTrap to be used with
          // jsdom for testing.
          displayCheck: disableTabbableDisplayCheck ? "none" : "full",
        },
      }}
    >
      <div aria-modal className={className} role="dialog">
        <button
          className="absolute active:text-neutral-500 hover:text-neutral-300 right-1 text-neutral-50 top-1 transition"
          onClick={handleClose}
        >
          <FontAwesomeIcon className="size-8 sm:size-12" icon={faXmark} />
          <AccessiblyHiddenText>Close</AccessiblyHiddenText>
        </button>
        <div className="container h-full">
          <Carousel
            className="h-full"
            images={sizedImages}
            showCaption
            slideDuration={null}
            transitionDuration={imageTransitionDuration}
          />
        </div>
      </div>
    </FocusTrap>
  );
}

export interface LightboxProps {
  className?: string;
  disableTabbableDisplayCheck?: boolean;
  imageIndex: number;
  images: CarouselProps["images"];
  imageTransitionDuration: CarouselProps["transitionDuration"];
  onClose: () => void;
  visibilityTransitionDuration: TransitionDuration | null;
}

function cleanUpPendingTimeout(timeoutId: MutableRefObject<number | null>) {
  return () => {
    if (timeoutId.current !== null) {
      window.clearTimeout(timeoutId.current);
    }
  };
}

function disableScrolling() {
  const bodyPaddingRight = parseFloat(
    getComputedStyle(document.body).paddingRight,
  );

  const scrollbarWidth =
    window.innerWidth - document.documentElement.clientWidth;

  document.body.style.overflow = "hidden";
  document.body.style.paddingRight = bodyPaddingRight + scrollbarWidth + "px";

  const enableScrolling = () => {
    document.body.style.overflow = "auto";
    document.body.style.paddingRight = bodyPaddingRight + "px";
  };

  return enableScrolling;
}

function makeVisible(
  isInitiallyVisible: boolean,
  setIsVisible: (isVisible: boolean) => void,
) {
  if (!isInitiallyVisible) {
    window.setTimeout(() => setIsVisible(true));
  }
}

function transformToSizedImage(image: ImageGallery["images"][number]) {
  const { height, width } = image;

  return {
    ...image,
    height: Math.min(height, 1_024),
    width: Math.min(width, 1_024),
  };
}
