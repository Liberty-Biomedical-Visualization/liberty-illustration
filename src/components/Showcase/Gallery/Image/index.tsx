import Image from "next/image";
import { useId } from "react";

import type { ImageData } from "@/lib/content";
import resolveClassNames from "@/lib/resolveClassNames";

export default function GalleryImage(props: Readonly<GalleryImageProps>) {
  const { imageData, onClick } = props;
  const { description, height, src, title, width } = imageData;

  const className = resolveClassNames(
    "overflow-hidden relative",
    props.className,
  );

  const captionId = useId();
  const titleContent = description ? `${title}:` : title;

  const descriptionContent = title ? (
    <>
      <br />
      {description}
    </>
  ) : (
    description
  );

  const caption = (
    <figcaption className={figcaptionClassName} id={captionId}>
      <p className={pClassName}>
        {title && <span className="sm:text-lg text-sm">{titleContent}</span>}
        {description && descriptionContent}
      </p>
    </figcaption>
  );

  const captionShouldBeShown = description || title;

  return (
    <button className={className} onClick={onClick}>
      {/* Chrome does not recognize figcaption as the name of figure. */}
      <figure aria-labelledby={captionId}>
        <Image
          alt={description ?? ""}
          className="aspect-square object-scale-down"
          height={height}
          src={src}
          title={title}
          width={width}
        />
        {captionShouldBeShown ? caption : <div className={shroudClassName} />}
      </figure>
    </button>
  );
}

export interface GalleryImageProps {
  className?: string;
  imageData: ImageData;
  onClick: () => void;
}

const figcaptionClassName =
  "absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition sm:active:bg-background/90 sm:hover:bg-background/80";

const pClassName =
  "absolute inset-0 opacity-0 p-2 sm:hover:opacity-100 text-left transition";

const shroudClassName =
  "absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition";
