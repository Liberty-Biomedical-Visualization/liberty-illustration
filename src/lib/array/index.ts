export { default as atChecked } from "./atChecked";
export { default as swap } from "./swap";
