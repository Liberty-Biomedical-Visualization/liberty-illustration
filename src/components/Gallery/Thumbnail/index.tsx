import Image from "next/image";
import { useId } from "react";

import type { ImageData } from "@/lib/content";
import resolveClassNames from "@/lib/resolveClassNames";
import truncateText from "@/lib/truncateText";

export default function Thumbnail(props: Readonly<ThumbnailProps>) {
  const { imageData, onClick } = props;
  const { description, title } = imageData;
  const thumbnailImageData = transformToThumbnail(imageData);

  const {
    description: thumbnailDescription,
    height,
    src,
    title: thumbnailTitle,
    width,
  } = thumbnailImageData;

  const className = resolveClassNames(
    "overflow-hidden relative",
    props.className,
  );

  const captionId = useId();
  const titleContent = description ? `${thumbnailTitle}:` : thumbnailTitle;

  const descriptionContent = title ? (
    <>
      <br />
      {thumbnailDescription}
    </>
  ) : (
    thumbnailDescription
  );

  const caption = (
    <figcaption className={figcaptionClassName} id={captionId}>
      <p className={pClassName}>
        {title && <span className="sm:text-lg text-sm">{titleContent}</span>}
        {description && descriptionContent}
      </p>
    </figcaption>
  );

  const captionShouldBeShown = description || title;

  return (
    <button className={className} onClick={onClick}>
      {/* Chrome does not recognize figcaption as the name of figure. */}
      <figure aria-labelledby={captionId}>
        <Image
          alt={description ?? ""}
          className="aspect-square object-scale-down"
          height={height}
          src={src}
          title={title}
          width={width}
        />
        {captionShouldBeShown ? caption : <div className={shroudClassName} />}
      </figure>
    </button>
  );
}

export interface ThumbnailProps {
  className?: string;
  imageData: Readonly<ImageData>;
  onClick: () => void;
}

/**
 * Maximum count of combined title and description characters before truncation.
 */
export const CAPTION_CHARACTER_LIMIT = 128;

function transformToThumbnail(image: Readonly<ImageData>): ImageData {
  const thumbnail: ImageData = {
    height: THUMBNAIL_SIZE,
    src: image.src,
    width: THUMBNAIL_SIZE,
  };

  let charactersRemaining = CAPTION_CHARACTER_LIMIT;

  for (const key of TEXT_KEYS) {
    const text = image[key];

    if (text && charactersRemaining > 0) {
      const truncatedText = truncateText(text, charactersRemaining);
      thumbnail[key] = truncatedText;
      charactersRemaining -= text.length;
    }
  }

  return thumbnail;
}

const TEXT_KEYS = ["title", "description"] as const;
const THUMBNAIL_SIZE = 256;

const figcaptionClassName =
  "absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition sm:active:bg-background/90 sm:hover:bg-background/80";

const pClassName =
  "absolute inset-0 opacity-0 p-2 sm:hover:opacity-100 text-left transition";

const shroudClassName =
  "absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition";
