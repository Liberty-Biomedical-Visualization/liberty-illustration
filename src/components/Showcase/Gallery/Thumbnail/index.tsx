import Image from "next/image";
import { useId } from "react";

import type { ImageData } from "@/lib/content";
import resolveClassNames from "@/lib/resolveClassNames";

export default function Thumbnail(props: Readonly<ThumbnailProps>) {
  const { imageData, onClick } = props;
  const thumbnailImageData = transformToThumbnail(imageData);
  const { description, height, src, title, width } = thumbnailImageData;

  const className = resolveClassNames(
    "overflow-hidden relative",
    props.className,
  );

  const captionId = useId();
  const titleContent = description ? `${title}:` : title;

  const descriptionContent = title ? (
    <>
      <br />
      {description}
    </>
  ) : (
    description
  );

  const caption = (
    <figcaption className={figcaptionClassName} id={captionId}>
      <p className={pClassName}>
        {title && <span className="sm:text-lg text-sm">{titleContent}</span>}
        {description && descriptionContent}
      </p>
    </figcaption>
  );

  const captionShouldBeShown = description || title;

  return (
    <button className={className} onClick={onClick}>
      {/* Chrome does not recognize figcaption as the name of figure. */}
      <figure aria-labelledby={captionId}>
        <Image
          alt={description ?? ""}
          className="aspect-square object-scale-down"
          height={height}
          src={src}
          title={title}
          width={width}
        />
        {captionShouldBeShown ? caption : <div className={shroudClassName} />}
      </figure>
    </button>
  );
}

export interface ThumbnailProps {
  className?: string;
  imageData: Readonly<ImageData>;
  onClick: () => void;
}

function transformToThumbnail(imageData: Readonly<ImageData>): ImageData {
  return { ...imageData, height: THUMBNAIL_SIZE, width: THUMBNAIL_SIZE };
}

const THUMBNAIL_SIZE = 256;

const figcaptionClassName =
  "absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition sm:active:bg-background/90 sm:hover:bg-background/80";

const pClassName =
  "absolute inset-0 opacity-0 p-2 sm:hover:opacity-100 text-left transition";

const shroudClassName =
  "absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition";
