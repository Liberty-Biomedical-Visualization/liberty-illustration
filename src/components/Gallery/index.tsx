"use client";

import { useState } from "react";

import { type ImageGallery } from "@/lib/content";
import resolveClassNames from "@/lib/resolveClassNames";

import Lightbox, { type LightboxProps } from "./Lightbox";
import Thumbnail, { type ThumbnailProps } from "./Thumbnail";

export default function Gallery(props: Readonly<GalleryProps>) {
  const {
    disableTabbableDisplayCheck,
    gallery,
    lightboxImageTransitionDuration,
    lightboxVisibilityTransitionDuration,
  } = props;

  const { images } = gallery;

  const [selectedImageIndex, setSelectedImageIndex] = useState<number | null>(
    null,
  );

  const thumbnails = images.map((imageData, index) =>
    transformToThumbnail(imageData, index, setSelectedImageIndex),
  );

  const lightboxIsShown = selectedImageIndex !== null;
  const handleLightboxClose = () => setSelectedImageIndex(null);
  const className = resolveClassNames(props.className, gridClassName);

  return (
    <>
      {lightboxIsShown && (
        <Lightbox
          disableTabbableDisplayCheck={!!disableTabbableDisplayCheck}
          imageIndex={selectedImageIndex}
          images={images}
          imageTransitionDuration={lightboxImageTransitionDuration}
          onClose={handleLightboxClose}
          visibilityTransitionDuration={lightboxVisibilityTransitionDuration}
        />
      )}
      <div className={className}>{thumbnails}</div>
    </>
  );
}

export interface GalleryProps {
  className?: string;
  disableTabbableDisplayCheck?: LightboxProps["disableTabbableDisplayCheck"];
  gallery: Readonly<ImageGallery>;
  lightboxImageTransitionDuration: LightboxProps["imageTransitionDuration"];
  lightboxVisibilityTransitionDuration: LightboxProps["visibilityTransitionDuration"];
}

function transformToThumbnail(
  imageData: ThumbnailProps["imageData"],
  index: number,
  onThumbnailClick: (index: number) => void,
) {
  const handleClick = () => onThumbnailClick(index);
  return <Thumbnail imageData={imageData} key={index} onClick={handleClick} />;
}

const gridClassName =
  "2xl:grid-cols-[repeat(6,minmax(0,256px))] gap-4 grid grid-cols-[repeat(3,minmax(0,256px))] lg:grid-cols-[repeat(4,minmax(0,256px))] max-w-fit xl:grid-cols-[repeat(5,minmax(0,256px))]";
