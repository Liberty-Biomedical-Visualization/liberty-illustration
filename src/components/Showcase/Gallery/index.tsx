import { type ImageGallery } from "@/lib/content";

import type { OnImageClick } from "..";
import Thumbnail, { type ThumbnailProps } from "./Thumbnail";

export default function Gallery(props: Readonly<GalleryProps>) {
  const { className, gallery, onImageClick } = props;
  const { images, title } = gallery;
  const thumbnails = images.map((imageData, index) =>
    transformToThumbnail(imageData, index, onImageClick),
  );

  return (
    <section className={className}>
      <h2 className="text-2xl">{title}</h2>
      <div className={gridClassName}>{thumbnails}</div>
    </section>
  );
}

export interface GalleryProps {
  className?: string;
  gallery: Readonly<ImageGallery>;
  onImageClick: OnImageClick;
}

function transformToThumbnail(
  imageData: ThumbnailProps["imageData"],
  index: number,
  onThumbnailClick: (index: number) => void,
) {
  const handleClick = () => onThumbnailClick(index);
  return <Thumbnail imageData={imageData} key={index} onClick={handleClick} />;
}

const gridClassName =
  "2xl:grid-cols-[repeat(6,minmax(0,256px))] gap-4 grid grid-cols-[repeat(3,minmax(0,256px))] lg:grid-cols-[repeat(4,minmax(0,256px))] max-w-fit xl:grid-cols-[repeat(5,minmax(0,256px))]";
