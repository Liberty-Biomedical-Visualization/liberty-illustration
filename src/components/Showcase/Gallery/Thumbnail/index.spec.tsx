import { describe, expect, it, jest } from "@jest/globals";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import {
  completeImage,
  describedImage,
  minimalImage,
  titledImage,
} from "@/lib/content/test-image-data";
import createTestRender from "@/lib/createTestRender";

import Thumbnail from ".";

describe("Thumbnail", () => {
  it("should display a figure", () => {
    renderThumbnail();
    const figure = screen.getByRole("figure");
    expect(figure).toBeVisible();
  });

  it("should not display a description when imageData has no description or title", () => {
    renderThumbnail({ imageData: minimalImage });
    const paragraph = screen.queryByRole("paragraph");
    expect(paragraph).toBeNull();
  });

  it("should display a figure with title as accessible name when imageData has title", () => {
    renderThumbnail({ imageData: titledImage });
    const figure = screen.getByRole("figure", {
      name: titledImage.title,
    });
    expect(figure).toBeVisible();
  });

  it("should display a figure with description as accessible name when imageData has description", () => {
    renderThumbnail({ imageData: describedImage });
    const figure = screen.getByRole("figure", {
      name: describedImage.description,
    });
    expect(figure).toBeVisible();
  });

  it("should display a figure with title and description as accessible name when imageData has title and description", () => {
    renderThumbnail({ imageData: completeImage });
    const figure = screen.getByRole("figure", {
      name: `${completeImage.title}: ${completeImage.description}`,
    });
    expect(figure).toBeVisible();
  });

  it("should display a button", () => {
    renderThumbnail();
    const button = screen.getByRole("button");
    expect(button).toBeVisible();
  });

  it("should display a button directing the user to view the image by its accessible name when available", () => {
    renderThumbnail({ imageData: completeImage });
    const button = screen.getByRole("button", {
      name: `${completeImage.title}: ${completeImage.description}`,
    });
    expect(button).toBeVisible();
  });

  it("should call onClick when the button is clicked", async () => {
    const onClick = jest.fn(() => {});
    renderThumbnail({ imageData: completeImage, onClick });

    const user = userEvent.setup();
    const button = screen.getByRole("button", {
      name: `${completeImage.title}: ${completeImage.description}`,
    });
    await user.click(button);

    expect(onClick).toHaveBeenCalled();
  });

  it("should pass className to the outermost element", () => {
    const className = "foo";
    const result = renderThumbnail({ className });
    expect(result.container.firstChild).toHaveClass(className);
  });

  it("should match the snapshot", () => {
    const result = renderThumbnail();
    expect(result.container).toMatchSnapshot();
  });
});

const renderThumbnail = createTestRender(Thumbnail, {
  imageData: minimalImage,
  onClick: () => {},
});
