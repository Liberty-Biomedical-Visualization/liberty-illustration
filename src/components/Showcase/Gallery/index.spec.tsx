import { describe, expect, it, jest } from "@jest/globals";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import assertIsDefined from "@/lib/assertIsDefined";
import {
  completeImageGallery,
  minimalImageGallery,
} from "@/lib/content/test-image-gallery-data";
import createTestRender from "@/lib/createTestRender";

import Gallery from ".";
import { atChecked } from "@/lib/array";

describe("Gallery", () => {
  it("should display a second level heading with the gallery title", () => {
    renderGallery({ gallery: minimalImageGallery });
    const heading = screen.getByRole("heading", {
      level: 2,
      name: minimalImageGallery.title,
    });
    expect(heading).toBeVisible();
  });

  it("should display images when the gallery has images", () => {
    renderGallery({ gallery: completeImageGallery });

    for (const image of completeImageGallery.images) {
      assertIsDefined(image.description);
      const img = screen.getByRole("img", { name: image.description });
      expect(img).toBeVisible();
    }
  });

  it("should should call onClick with the index of each image when clicked", async () => {
    const onImageClick = jest.fn();
    renderGallery({ gallery: completeImageGallery, onImageClick });
    const user = userEvent.setup();

    for (let index = 0; index < completeImageGallery.images.length; ++index) {
      const image = atChecked(completeImageGallery.images, index);
      assertIsDefined(image.description);
      const img = screen.getByRole("img", { name: image.description });
      await user.click(img);
      expect(onImageClick).toHaveBeenCalledWith(index);
    }
  });

  it("should pass className to the outermost element", () => {
    const className = "foo";
    const result = renderGallery({ className });
    expect(result.container.firstChild).toHaveClass(className);
  });

  it("should match the snapshot", () => {
    const result = renderGallery();
    expect(result.container).toMatchSnapshot();
  });
});

const renderGallery = createTestRender(Gallery, {
  gallery: minimalImageGallery,
  onImageClick: () => {},
});
