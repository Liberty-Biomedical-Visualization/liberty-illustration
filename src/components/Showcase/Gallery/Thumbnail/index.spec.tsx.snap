// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should match the snapshot 1`] = `
<div>
  <button
    class="overflow-hidden relative"
  >
    <figure
      aria-labelledby=":r9:"
    >
      <img
        alt=""
        class="aspect-square object-scale-down"
        data-nimg="1"
        decoding="async"
        height="256"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Ffoo.com%2Fminimal.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Ffoo.com%2Fminimal.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Ffoo.com%2Fminimal.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="256"
      />
      <div
        class="absolute inset-0 active:bg-accent-200/50 hover:bg-accent-50/50 transition"
      />
    </figure>
  </button>
</div>
`;
